### ISSUE TITLE
identical dimensions for svg and png background / or excempt specific pages from svg-format

### ISSUE THREAD

**Issue by HeimMatthias** [coolwanglu/pdf2htmlEX#642]
Hey,

We've been fiddling around with pdf2htmlEX for the epub3 conversion of complicated modern layouts. In our books, there are pages with massively complicated vector images and the `--svg-node-count-limit ` option has been massively helpful in getting useful output.

However, some pages still fail to render. Sometimes the generated SVG isn't accurate, sometimes the generated SVG makes browsers crash (or, in one case, OSX). This isn't a pdf2htmlEX-problem, per se, I suppose Cairo is to blame.

But in all cases, PNG-output works flawlessly! So what we would like to do, is to replace the problematic SVGs with PNGs. But we found that the dimensions of the SVGs differ from the PNGs, so that it isn't easily possible to just replace the file and the `src`-attribute in the HTML. It seems that blank space around png-background is cropped. This may save a couple of bytes, but it would be far more helpful to have interchangeable background files than to save a little bit of well-compressed space.

Alternatively, at least for our very specific scenario, we would appreciate an option that enforces PNG-backgrounds for specific pages, so that one could handtune the output until Cairo and webkit catch up with our standards. Something like `--force-png-background 3,6,18` maybe.

**Comment by duanyao** [comment]
Yeah, I am aware of the limitation of `--svg-node-count-limit` and `--force-png-background` can be a rescue. But I can't promise to implement this feature any time soon. Patch is welcome!

P.S. pdf2htmlEX use poppler's cairo backend to produces SVG, so if the SVG background looks wrong, you can try poppler's `pdf2cairo -svg` tool to reproduce the problem and report it at https://bugs.freedesktop.org/ (under product poppler or cairo).
